### DISCUSSION TITLE
Matériel électronique

### DISCUSSION THREAD

**Discussion by duayfabi** [FredM67/PVRouter-3-phase#78] [Q&A]
Bonjour Frédéric,

Tout d'abord, un grand merci pour votre travail et surtout pour son partage, c'est une véritable mine d'or !

Une question, certainement toute bête. Le matériel de base, je pense à la carte mère et aux autres componsants doivent bien être achetés sur le nouveau site de vente de mk2pvrouter ?

Il ne me semble pas avoir vu de schematics dans votre repo.. je sais que le créateur oginiel de mk2pvrouter en faisait commerce, il semble naturel qu'il n'ai pas donné accès au schema de ses PCB.

Ce n'est pas dérangant, c'est simplement pour être certain de prendre le bon matériel au bon endroit.

Merci d'avance.

Fabien Duay

**Comment by FredM67** [comment]
Salut Fabien,
Merci merci ;-)
Les schémas électroniques et pcb ne sont pas disponibles pour le moment, c'est vrai. Il n'y a rien de secret, Robin fournissait un schéma de principe. Nous sommes un peu hésitant à publier ces données essentiellement pour des raisons de sécurité électrique.
Je vois régulièrement des personnes acheter des gradateurs chez AliExpress, des trucs qui sont extrêmement dangereux si on les utilise tel quel.
Les composants fournis par mk2pvrouter.com sont tous de qualité et proviennent de revendeurs comme Farnell ou Mouser.
La carte mère a une partie sous 230V, tout comme chaque carte de sortie... il ne faut donc pas faire n'importe quoi !
Quand le schéma sera stable (j'y ai apporté plusieurs fois des modifications pour améliorer certains points), il sera publié au sein de la doc qui est elle aussi en cours de réécriture complète.

Oui, il "faut" tout commander sur le site mk2pvrouter.com, voire directement par mail à Quentin pour du sur-mesure (plus de 3 sorties, 1 ou plusieurs sorties tout-ou-rien, multiboitiers, ...).

Fred

**Comment by duayfabi** [comment]
Hello Frédéric,

Merci ta réponse, je vais donc regarder directement avec Quentin.

Je ne sais pas comment prendre la phrase "quand le shéma sera stable" et l'associer au fait d'utiliser mk2pvrouter en prod chez moi sur du triphasé. J'en discuterai directement avec Quentin :)

Si jamais, je serais intéressé a aider le projet. J'aurais trouvé intéressant de remplacer l'atmega328 par un esp32 par exemple, ça ouvrirait d'intéressantes perspectives.. enfin bref, si ça peut être intéressant..

Fabien

**Reply by FredM67** [reply]
Lol, les modifications concernent des "broutilles"... genre récemment j'ai doublé certains connecteurs...
Le coeur du système est stable depuis longtemps !
Mon routeur tri, le même, fonctionne depuis 4 ans !!

Remplacer par un esp32 n'est pas vraiment faisable avec le tri... il faudrait plutôt passer sur une puce complètement nouvelle, genre celle utilisée dans l'EmontTx V4 ! En tri, il faut 6 ADC !!

Mais oui, très volontiers, tu peux participer aux évolutions... au besoin, je peux te rajouter dans les contributeurs sur les 2 dépôts pour le code !

**Reply by FredM67** [reply]
[3phaseDiverter.pdf](https://github.com/FredM67/PVRouter-3-phase/files/13219665/3phaseDiverter.pdf)
[Output_stage.pdf](https://github.com/FredM67/PVRouter-3-phase/files/13219675/Output_stage.pdf)

**Reply by duayfabi** [reply]
Merci pour les schémas!
J'ai contacté Amandine et Quentin, j'ai dans l'idée de construire déjà mon premier router et de voir ensuite ce qui me manquerait.
A vue de nez, le système RF me serait utile pour avoir de la data avec mon système domotique (home assistant), mais j'ai peur de la fiabilité de la transmission des données.
Un pas après l'autre, je vais déjà commencer par construire mon premier router et voir ensuite..

**Reply by FredM67** [reply]
J'ai déjà en beta un esp32 ainsi que l'interface graphique qui va avec pour piloter le routeur, genre période de vacances,...

**Reply by duayfabi** [reply]
Ah sympa, donc toujours avec l'atmega pour les ADC mais avec une liaison de comm. entre l'atmega et l'esp32 ?

**Reply by FredM67** [reply]
Yes, exactement!
C'est aussi pour cela que les nouveaux PCB peuvent avoir une double alim.
L'arduino est en 3.3V, pour l'esp32, souvent il faut alimenter en 5V...

**Reply by FredM67** [reply]
http://82.64.56.179:2525/

tu peux jouer avec, rien n'est branché derrière !

**Reply by duayfabi** [reply]
Salut Fred,

Super inéressant. Je vois qu'une intégration mqtt a été pensée, ce qui est top.
Je ne connaissais pas openenergymonitor, tu utilises emontx ?

Je me demande à quel point il serait malin de complètement découpler les mesures d'énergie et le contrôle des appareils.

J'ai aussi un chargeur ev dont je veux limiter le courant selon ma production solaire. Difficile de le faire avec mk2pvrouter, l'électronique de ma wallbox ne le supporterait certainement pas. Je peux le faire en utilisant le protocole modbus mais je dois forcément connaître le surplus de production et comme mk2pvrouter calcul ce qui est injecté sur le réseau.. serait bien qu'il me le communique.

D'avoir un appareil de mesure complètement découplé permettrait d'avoir un sytème plus modulable.

Je vais plus me renseigner sur emontx.. me manque des infos pour me faire une bonne idée de ses possibilités.

**Reply by FredM67** [reply]
La nouvelle puce utilisée dans EmonTX version 4 dispose de beaucoup plus d'adc.
Notre but à l'avenir mais il faudra trouver le temps c'est de migrer vers ce genre de hardware.

**Reply by duayfabi** [reply]
Plus d'adc et plus de précision dans les mesures..

J'ai quelques mois devant moi avant que mes panneaux solaire arrivent, je peux voir ce que je peux faire, j'ai téléchargé ton repo, je suis en train de me faire une idée du projet..

Je veux aussi piloter mon chargeur ev à travers une communication modbus afin de limiter le courant utilisé pour charger ma voiture, je pensais le faire avec evcc.io mais dans ce cas-là, difficile de gérer les priorités entre mon chauffe-eau et evcc.. Dans les faits, le chauffe eau serait toujours prioritaire et la charge du véhicule passerait toujours après.

J'ai quelques idées qui ne sont pas parfaites et qui ne permettent pas de charger la voiture ET de chauffer mon eau, c'est toujours soit l'un soit l'autre. Si tu as une inspiration, je suis volontiers preneur.

**Reply by FredM67** [reply]
Ce qui est important, c'est surtout la qualité de l'ADC ainsi que le nombre.
Oui ce n'est pas facile de combiner les 2, et de toute facon, je pense qu'il n'y aura jamais vraiment une solution "parfaite" pour chacun.
Certains voudront avoir d'abord de l'eau chaude, d'autre d'abord leur voiture chargée, ...
Il faut arriver à trouver un système de communication pour piloter le routeur.. et là aussi, il y a un million de possibilité, entre ModBus, M-Bus, et tous les autres protocoles qui existent !
